### ISSUE TITLE
Document opinion on focus ring suppression

### ISSUE THREAD

**Issue by johanstromqvist** [Shopify/polaris-react-archive#5504]
**Problem**
The purpose of the focus ring is to clearly show what element is ready for interaction and, when navigating with keyboard, direct users' attention to the element that focus moves to.

When navigating with pointing devices such as mouse or trackpad, the focus ring is often redundant and creates visual noise and distractions instead of adding clarity. This has negative impact on both accessibility and perceived performance.

![](https://camo.githubusercontent.com/10f5d1d0c0321f7096feff8bc348d6e9d7808d4ae361660653407ef92cafcf63/68747470733a2f2f73637265656e73686f742e636c69636b2f31312d32312d36726332672d706c676e392e676966)

**Suggested approach**
- [ ] Research considerations for focus ring suppression
- [ ] Document our opinion on when and why we should do it
- [ ] Implement in our primitives and provide examples

**Comment by svinkle** [comment]
The [`:focus-visible` CSS pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) is [well supported by major browsers](https://caniuse.com/?search=focus-visible). This is [used in our free themes](https://github.com/Shopify/dawn/blob/d0463c6bc72548a8f4bbfc42dcbaf7f135683190/assets/base.css#L788) to achieve the effect described here.

What's great about this technique is it gives designers space to be innovative in creating a custom, high contrast focus ring while getting out of the way for sighted mouse users. Focus indicators are to be highly visible and contrast well against adjacent backgrounds (contrast ratio of at least `3:1` between its focused and unfocused state – [1.4.11: Non-text Contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html).)

Please note, for any area where it's assumed the user is going to provide keyboard input (such as an `input type="text"` element), the browser **will** draw a focus indicator. This is articulated in [the CSS spec](https://drafts.csswg.org/selectors-4/#the-focus-visible-pseudo) in the second half of the opening statement:

> The :focus-visible pseudo-class applies while an element matches the :focus pseudo-class and the UA determines via heuristics that the focus should be made evident on the element.
>
> For example, UAs typically display focus indicators on text fields any time they’re focused, to draw attention to the fact that keyboard input will affect their contents.
>
> On the other hand, UAs typically only display focus indicators on buttons when they were focused by a keyboard interaction (such as tabbing through the document)—because it’s not always immediately obvious where the focus has gone after such an interaction, but is sufficiently self-evident when the button was focused by more obviously-targetted interactions, like clicking on the button with a mouse pointer.
